python/thunder/rdds/data.py

```diff
@@ -111,6 +111,129 @@ def take(self, *args, **kwargs):
         """
         return self.rdd.take(*args, **kwargs)
 
+    def get(self, key):
+        """Returns a single value matching the passed key, or None if no matching keys found
+
+        If multiple records are found with keys matching the passed key, a sequence of all matching
+        values will be returned. (This is not expected as a normal occurance, but could happen with
+        some user-created rdds.)
+        """
+        filteredVals = self.rdd.filter(lambda (k, v): k == key).values().collect()
```

Comment:
Could also use our `filterOnKeys` method, would simplify the lambda and avoid dereferencing the rdd, right?